src/information_schema.rs

```diff
@@ -0,0 +1,795 @@
+//! `information_schema` pattern. Uses live querying - metadata is fetched fresh from
+//! the catalog database on every query execution.
+//!
+//! # Available Tables
+//!
+//! - `information_schema.snapshots` - All snapshots in the catalog
+//! - `information_schema.schemata` - Schemas at current snapshot
+//! - `information_schema.tables` - Tables across all schemas at current snapshot
+//! - `information_schema.columns` - Columns for all tables
+//! - `information_schema.files` - Data files for all tables
+//!
+//! # Usage
+//!
+//! ```sql
+//! -- List all snapshots
+//! SELECT * FROM ducklake.information_schema.snapshots;
+//!
+//! -- List schemas
+//! SELECT * FROM ducklake.information_schema.schemata;
+//!
+//! -- List tables
+//! SELECT * FROM ducklake.information_schema.tables WHERE schema_name = 'public';
+//! ```
+
+use std::any::Any;
+use std::sync::Arc;
+
+use arrow::array::{ArrayRef, BooleanArray, Int64Array, RecordBatch, StringArray};
+use arrow::datatypes::{DataType, Field, Schema, SchemaRef};
+use datafusion::catalog::{SchemaProvider, Session};
+use datafusion::datasource::TableProvider;
+use datafusion::datasource::memory::MemTable;
+use datafusion::error::Result as DataFusionResult;
+use datafusion::logical_expr::TableType;
+use datafusion::physical_plan::ExecutionPlan;
+
+use crate::metadata_provider::MetadataProvider;
+
+/// Live table provider for snapshots - queries metadata on every scan
+#[derive(Debug)]
+pub struct SnapshotsTable {
+    provider: Arc<dyn MetadataProvider>,
+    schema: SchemaRef,
+}
+
+impl SnapshotsTable {
+    pub fn new(provider: Arc<dyn MetadataProvider>) -> Self {
+        let schema = Arc::new(Schema::new(vec![
+            Field::new("snapshot_id", DataType::Int64, false),
+            Field::new("timestamp", DataType::Utf8, true),
+        ]));
+        Self {
+            provider,
+            schema,
+        }
+    }
+
+    fn query_snapshots(&self) -> DataFusionResult<RecordBatch> {
+        let snapshots = self
+            .provider
+            .list_snapshots()
+            .map_err(|e| datafusion::error::DataFusionError::External(Box::new(e)))?;
+
+        let snapshot_ids: ArrayRef = Arc::new(Int64Array::from(
+            snapshots.iter().map(|s| s.snapshot_id).collect::<Vec<_>>(),
+        ));
+
+        let timestamps: ArrayRef = Arc::new(StringArray::from(
+            snapshots
+                .iter()
+                .map(|s| s.timestamp.as_deref())
+                .collect::<Vec<_>>(),
+        ));
+
+        RecordBatch::try_new(self.schema.clone(), vec![snapshot_ids, timestamps])
+            .map_err(|e| datafusion::error::DataFusionError::ArrowError(Box::new(e), None))
+    }
+}
+
+#[async_trait::async_trait]
+impl TableProvider for SnapshotsTable {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn schema(&self) -> SchemaRef {
+        self.schema.clone()
+    }
+
+    fn table_type(&self) -> TableType {
+        TableType::Base
```

Comment:
```suggestion
        TableType::View
```

This should be a view, given that it's backed by a query. This is somewhat just semantics, but DataFusion's built-in information_schema table uses View, so probably makes sense to match the convention.